src/upkeep.rs

```diff
@@ -379,6 +379,9 @@ pub async fn check_health(
     mut health_reporter: HealthReporter,
 ) -> Instant {
     let now = Instant::now();
+    if config.health_check_killswitched {
+        return now;
```

Comment:
Should you set the status to "serving" in this case ?
```
health_reporter
            .set_service_status(SERVICE_NAME, ServingStatus::Serving)
            .await;
```

So that, if the status is "not serving" and somebody triggers the killswitch k8s would not interpret the pod as being in not serving state.